spritecontrol.cpp

```diff
@@ -391,6 +392,247 @@ std::map<PatternName, GlobalPlayer::MoveFunction> GlobalPlayer::move_functions {
 			glEnd();
 		}
 	}},
+	{ Boids, [](Sprites *sprites, Context *ctx, std::function<double(Id)> get_offset) {
+		const static double SCREEN_SIZE = (double) ((long long) ctx->rect().bottom * ctx->rect().right);
+		const static double STRETCH_RATIO = (double) (ctx->rect().bottom) / ctx->rect().right;
+		const static double SEPARATION_Y_RADIUS = (6.0 / (cfg[Cfg::SpriteCount] / 1.5 + 40.0)) * std::pow(SCREEN_SIZE / (1080LL * 1920LL) / 3.0 + 0.7, 1.1);
+		const static double SEPARATION_X_RADIUS = SEPARATION_Y_RADIUS * STRETCH_RATIO;
+		const static double VISION_Y_RADIUS = (10.0 / (cfg[Cfg::SpriteCount] / 15.0 + 20.0)) * std::pow(SCREEN_SIZE / (1080LL * 1920LL) / 3.0 + 0.7, 1.1);
```

Comment:
Did I really hard code this shit for 1920x1080? what a disaster? I truly did not give a shit.